src/utils/getUrlByRequestOptions.ts

```diff
@@ -100,31 +93,8 @@ function getAuthByRequestOptions(
   }
 }
 
-/**
- * Returns true if host looks like an IPv6 address without surrounding brackets
- * It assumes any host containing `:` is definitely not IPv4 and probably IPv6,
- * but note that this could include invalid IPv6 addresses as well.
- */
-function isRawIPv6Address(host: string): boolean {
-  return host.includes(':') && !host.startsWith('[') && !host.endsWith(']')
-}
-
-function getHostname(host: string, port?: number): string {
-  const portString = typeof port !== 'undefined' ? `:${port}` : ''
-
-  /**
-   * @note As of Node >= 17, hosts (including "localhost") can resolve to IPv6
-   * addresses, so construct valid URL by surrounding the IPv6 host with brackets.
-   */
-  if (isRawIPv6Address(host)) {
-    return `[${host}]${portString}`
-  }
-
-  if (typeof port === 'undefined') {
-    return host
-  }
-
-  return `${host}${portString}`
+function getHostname(host: string): string {
```

Comment:
Let's revert this back, there's most history behind this than first catches the eye. 

A bit of context: it's possible for the `ClientRequest` options to have all sorts of combinations of `host`, `hostname` and `port`. Any can be undefined and in some cases must be inferred from the other value so the interceptor would compose the right URL. 

There are some libraries out there can use `ClientRequest` rather. . . unconventionally. We must still support those usages, thus such a bullet-proof resolution for host/hostname.

Reply:
Sure, can you please be more specific?
What do we need to revert?

If the `host` is undefined, should we set the `hostname` as `host`?

Reply:
We should keep those two helper functions as they were before: `getHostFromRequestOptions` and `getHostname`. I don't think we should change them, unless there's an issue in what they return.

Reply:
> unless there's an issue in what they return

```js
if (hostname != null) {
    return hostname.replace(/:\d+$/, '')
  }
```
This deletes the last part of the ipv6: `2607:f0d0:1002:51::4` become to `2607:f0d0:1002:51:`

Reply:
Nice! So there's a bug in the regexp, let's fix that 👀 

Would it be beneficial if we prioritized `host`, and if it exists, don't even bother to infer it from `hostname`?

Reply:
> Nice! So there's a bug in the regexp, let's fix that 👀

I think we don't need the regex at all. The hostname is a "subset" of the host. we don't need to remove the port from the hostname, because it doesn't have one.

I think:

1. if host exists: return host
2. if hostname exists: return hostname
3. otherwise: DEFAULT

Reply:
😆 I think that line meant `host`. Uhh, I need to re-read all of this... 

I think we could keep both utilities so we don't change much and can refactor this separately. For now, the intention behind those functions is the following:

### `getHostByRequestOptions`

```js
if (options.host) {
  return options.host
}

if (options.hostname) {
  if (options.port) {
    return `${options.hostname}:${options.port}`
  }

  return options.hostname
}

return null
```

### `getHostname`

```js
if (options.hostname) {
  return options.hostname
}

if (options.host) {
  // Check the presence of the port, and if it's present,
  // remove it from the host, returning a hostname.
  return hostWithoutPort(options.host)
}
```

Reply:
I pushed a commit.

I fixed another confusion in `getPortByRequestOptions`. PTAL. I'm not sure if it's ipv6-proof.
Also, do we want `getHostByRequestOptions` and `getHostname` to return `null` or `DEFAULT_HOST`?

Reply:
I'd say they should return `undefined` if there was no value set in the options. Then, the actual usage surface of those functions can have fallbacks to `DEFAULT_*`, if needed.